pygmt/exceptions.py

```diff
@@ -51,3 +54,57 @@ class GMTImageComparisonFailure(AssertionError):  # noqa: N818
     """
     Raised when a comparison between two images fails.
     """
+
+
+class GMTValueError(GMTError):
+    """
+    Raised when an invalid value is passed to a function/method.
+
+    Parameters
+    ----------
+    value
+        The invalid value.
+    description
+        The description of the value.
+    choices
+        The valid choices for the value.
+    reason
+        The detailed reason why the value is invalid.
+
+    Examples
+    --------
+    >>> raise GMTValueError("invalid")
+    Traceback (most recent call last):
+        ...
+    pygmt...GMTValueError: Invalid value: 'invalid'.
+    >>> raise GMTValueError("invalid", description="constant name")
+    Traceback (most recent call last):
+    ...
+    pygmt...GMTValueError: Invalid constant name: 'invalid'.
+    >>> raise GMTValueError("invalid", choices=["a", "b", 1, 2])
+    Traceback (most recent call last):
+        ...
+    pygmt...GMTValueError: Invalid value: 'invalid'. Expected one of: 'a', 'b', 1, 2.
+    >>> raise GMTValueError("invalid", choices=["a", 0, True, False, None])
+    Traceback (most recent call last):
+        ...
+    pygmt...GMTValueError: Invalid .... Expected one of: 'a', 0, True, False, None.
+    >>> raise GMTValueError("invalid", reason="Explain why it's invalid.")
+    Traceback (most recent call last):
+        ...
+    pygmt...GMTValueError: Invalid .... Explain why it's invalid.
+    """
+
+    def __init__(
+        self,
+        value: Any,
+        description: str = "value",
```

Comment:
From your changes, it seems like the `value` parameter is used in a positional-only way. Do you want to enforce it with https://peps.python.org/pep-0570/#syntax-and-semantics, or still let `value` be a 'positional_or_keyword_parameters'?

```suggestion
        value: Any,
        /
        description: str = "value",
```